Source/WebKit/UIProcess/API/C/WKContextPrivate.h

```diff
@@ -31,6 +31,8 @@
 
 #if defined(WIN32) || defined(_WIN32)
 typedef int WKProcessID;
+#elif PLATFORM(HAIKU)
+typedef int32_t WKProcessID;
```

Comment:
This is not right, pid_t should be used for WKProcessID. Adjust the places where it is encoded/decoded if needed.

Reply:
As far as i know WTF::ProcessID should be same as WKProcessID . We gave WTF::ProcessID as int32_t explicitly so there is ambuiguity rised in function definitions because WKProcessID is of pid_t.
So u think setting WKPRocessID to int32_t explicitly like we did for WTF::processid is viable solution?

Reply:
I think it would be better to keep pid_t and use the toPid/toProcessId functions I added to convert between the two types?

Reply:
I was wondering the same 😕 . We need not use converting right as having int32_t is simple and viable solution will not introduce much merge conflicts what say?

Reply:
In this case we have found a real problem in WebKit code, and we will send them the change. So we can make more modifications, and once they have accepted the patch, there will be no merge conflict.

Reply:
So can i right away explicitly convert to int32_t the WKProcessID?
I asked the webkit-dev still now there was no response so when they suggest something better we better adapt that

Reply:
They already replied to my mail previously and you asked the same question again. Surely they will find this boring and will not reply a second time.

As I said, I added the toPid/toProcessId functions so we can use them to convert between WKProcessID (pid_t) and WTF::ProcessID (int32_t) and back